pkg/status/status.go

```diff
@@ -42,17 +42,21 @@ const (
 	ConditionTypeIncorrectNUMAResourcesOperatorResourceName = "IncorrectNUMAResourcesOperatorResourceName"
 )
 
-func Update(ctx context.Context, client k8sclient.Client, rte *nropv1alpha1.NUMAResourcesOperator, condition string, reason string, message string) error {
+func Update(ctx context.Context, client k8sclient.Client, rte *nropv1alpha1.NUMAResourcesOperator, condition string, reason string, message string) (bool, error) {
 	conditions := NewConditions(condition, reason, message)
+	for index := 0; index < len(rte.Status.Conditions); index++ {
```

Comment:
since we need to iterate over all the conditions anyway, and since the `semantic.DeepEqual` has no choice but do the same (in a generic way, but still), I'm wondering if we should not just add an explicit equality check which does NOT consider timestamps - but does consider everything else.
That will also enable us to reuse the `conditions` we build on line 46 instead of re-creating on line 54

Reply:
Yep, I was also thinking about that . I  went with this one to keep using `semantic.DeepEqual` but I can give it a try...

Reply:
looking around ... google go-cmp package has an [`IgnoreField`](https://pkg.go.dev/github.com/google/go-cmp/cmp/cmpopts#IgnoreFields) option that could be useful here. 

Code will get something like 
`ret := cmp.Equal(conditions, rte.Status.Conditions, cmpopts.IgnoreFields(metav1.Condition{}, "LastTransitionTime"))
`

More info: https://github.com/google/go-cmp

Reply:
we already depend on `go-cmp`, so this is a good option.
Looking at `metav1.Condition`, the struct is pretty simple:
```
type Condition struct {
	Type string `json:"type" protobuf:"bytes,1,opt,name=type"`
	Status ConditionStatus `json:"status" protobuf:"bytes,2,opt,name=status"`
	ObservedGeneration int64 `json:"observedGeneration,omitempty" 
	LastTransitionTime Time `json:"lastTransitionTime" protobuf:"bytes,4,opt,name=lastTransitionTime"`
	Reason string `json:"reason" protobuf:"bytes,5,opt,name=reason"`
	Message string `json:"message" protobuf:"bytes,6,opt,name=message"`
}
```
we should also ignore `ObservedGeneration`.
An explicit function to check equality seems also still very much on the table.

Reply:
Ok. Both ways seem ok to me so if any have a preference between go-cmp and a explicit function I will take it.

Reply:
let's start with `go-cmp`